Change class name clearly

SeulGeek opened the pull request.

Commit:
Change class name clearly

EldredRoer commented on app/src/main/java/com/app/foodforyou/model/DietListResponse.java:
```diff
@@ -3,7 +3,7 @@
 // Response API
-public class RecommendDietListResponse {
+public class DietListResponse {
     private int cntntsNo; // 컨텐츠 번호
```
Please write descriptive variable names to provide more clarity :)

SeulGeek replied:
Thank you for your feedback! I'm trying to change code using Retrofit. then object's variable name should be same with API response's variable name. so I should keep this now.. Instead, I'll add English comment. :)

EldredRoer commented on app/src/main/java/com/app/foodforyou/view/DietListActivity.java:
```diff
@@ -105,9 +104,9 @@ public void run() {
-                PreferenceManager.setString(mContext, "cntntsNo", String.valueOf(dataManager.getRecommendDietListResponses().get(position).getCntntsNo()));
+            public void onDietItemClick(DietListAdapter.ViewHolder holder, View view, int position) {
+                Intent intent = new Intent(getApplicationContext(), FoodListActivity.class);
+                PreferenceManager.setString(mContext, "cntntsNo", String.valueOf(dataManager.getDietListResponses().get(position).getCntntsNo()));
```
cntntsNo should be static final, not hard coded

SeulGeek replied:
Thanks! Which is better to make clean code?

1. Declare static final string variable in same class (ex. public static final String CONTENTS_NO)
2. Make to string value (ex. R.string.contentsNo)

EldredRoer replied:
number 1 is recommended

EldredRoer commented on app/src/main/java/com/app/foodforyou/view/DietListActivity.java:
```diff
@@ -137,7 +136,7 @@ private void getRecommendDietListResponse() {
                     startTag = xpp.getName();
                     if (startTag.equals("item")) {
```
No hard coding please

EldredRoer commented on app/src/main/java/com/app/foodforyou/view/MainActivity.java:
```diff
@@ -154,7 +154,7 @@ private void getMainCategoryResponse() {
 
     private void onClickCategory() {
-        intent = new Intent(getApplicationContext(), RecommendDietListActivity.class);
+        intent = new Intent(getApplicationContext(), DietListActivity.class);
```
application context is tied to Application and its lifecycle. So should be used in such cases:

- Load Resource Values
- Start a Service
- Bind to a Service
- Send a Broadcast
- Register BroadcastReceiver

So for intent, use activity instead.

SeulGeek replied:
Thank you for your explaining in detail. :) 
I've learned about service but I still confused. so could you make sure the way I thought is correct? 

- Load Resource Values -> this should be run using **getMainCategoryResponse** method in **MainActivity**
- Start a Service -> Make new Service class (ex. GetMainCategoryService)
- Bind to a Service -> In new service class, it should be bind with DietListActivity.

If I miss or wrong something, please let me know. Thank you!

EldredRoer replied:
Ahh small misunderstanding Stella. I was just explaining the cases when applicationContext should be used :)

Basically if you dont need to use service or broadcast receiver in app then no need to use applicationContext. Just simple context or activity is enough.

Good article to explain more:

https://blog.mindorks.com/understanding-context-in-android-application-330913e32514

SeulGeek replied:
got it thank you! I think part of network is related with service. I'll make new issue and commit.
Thank you! 👍

EldredRoer commented on app/src/main/java/com/app/foodforyou/view/FoodListAdapter.java:
```diff
@@ -80,7 +80,7 @@ public void onClick(View v) {
 
-        void onBind(RecommendDietDetail data) {
+        void onBind(FoodRecipeDetail data) {
             if (!TextUtils.isEmpty(data.getRtnImageDc())) {
```
Please use more descriptive method name :)

EldredRoer commented on app/src/main/java/com/app/foodforyou/view/FoodListActivity.java:
```diff
@@ -93,10 +93,10 @@ public void run() {
             @Override
             public void onFoodItemClick(View view, int position) {
-                Intent intent = new Intent(getApplicationContext(), FoodRecipeDetails.class);
+                Intent intent = new Intent(getApplicationContext(), FoodRecipeDetailsActivity.class);
```
activity please not getApplicationContext

EldredRoer replied:
This is still using getApplicatiionContext, please change to activity context :)

SeulGeek replied:
Thank you! Is this correct what you want?

` Intent intent = new Intent(getActivity(), FoodRecipeDetailsActivity.class);`

If it is correct, it seems impossible. there are only 3 methods.. could you let me know another way? :)
<img width="704" alt="image" src="https://user-images.githubusercontent.com/64777031/94443657-6395dd00-01e0-11eb-9ff0-57a296b80166.png">

EldredRoer replied:
Since its activity then can just use 'this', getActivity will be used in fragment :)

SeulGeek replied:
ahh got it. thank you! 
In this activity, it couldn't set "this" because intent is in the setOnFoodItemClickListener. 
so the intent variable should be declared global. but I've already set "mContext = this" so I used "mContext" instead of "this".
I fixed it. please look at it again. thanks a lot 👍

EldredRoer replied:
Ahh in the case where context is inside a listener then use "activity.this", in this case FoodRecipeDetailsActivity.this :)

Commit:
- Declare static final string variable in same class
- Use descriptive method name
- Use "this" instead of "getApplicationContext()" in the case of
activity

SeulGeek commented:
[Summary]

**DONE**
- Change class name clearly
- Declare static final string variable in same class (not hard coding) 
- Use descriptive method name
- In case of activity, please not getApplicationContext

**TO DO**
- Make new issue to separate network feature using service

SeulGeek commented:
@EldredRoer please check new commit when you have time. If it is ok, I'll merge it. :)

EldredRoer reviewed (approved):
Approved but change the context for next PR :)
